drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchyInflater.java

```diff
@@ -190,6 +212,20 @@ public static GenericDraweeHierarchyBuilder updateBuilder(
         }
       } finally {
         gdhAttrs.recycle();
+
+        if (android.os.Build.VERSION.SDK_INT >= 17 && context.getResources().getConfiguration()
+            .getLayoutDirection() == View.LAYOUT_DIRECTION_RTL) {
+          roundTopLeft = roundTopLeft && roundTopEnd;
+          roundTopRight = roundTopRight && roundTopStart;
+          roundBottomRight = roundBottomRight && roundBottomStart;
+          roundBottomLeft = roundBottomLeft && roundBottomEnd;
+        } else {
+          roundTopLeft = roundTopLeft && roundTopStart;
```

Comment:
Thanks for the pull request @adamc01 !

Overall it looks great but I think we have an issue where when it comes to combining roundXstart/end with roundXleft/right.

If you compare this to how setting margins or padding for RTL is done:

**Set left margin that should become right margin for RTL locales**
android:layout_marginStart="10dp"
android:layout_marginLeft="10dp"

While here its reversed:

roundTopLeft = roundTopLeft && roundTopEnd;

In the case of RTL layout direction on Android 4.2+, the top left corner should be rounded if

roundTopEnd==true && roundTopRight is either true or not set

Let me know if this is clear and if it makes sense to you. Thanks!

Reply:
On API 17+ force RTL mode (make sure `android:supportsRtl="true"` is set in the AndroidManifest) and if you do the following:
```
      <com.facebook.drawee.view.SimpleDraweeView
          android:id="@+id/drawee_some_rtl"
          style="@style/DraweeRoundedCorners.Drawee"
          app:roundedCornerRadius="16dp"
          android:layout_marginStart="50dp"
          android:layout_marginLeft="50dp"
          />
```

then Start does indeed override Left. However if you do:
```
      <com.facebook.drawee.view.SimpleDraweeView
          android:id="@+id/drawee_some_rtl"
          style="@style/DraweeRoundedCorners.Drawee"
          app:roundedCornerRadius="16dp"
          android:paddingStart="50dp"
          android:paddingLeft="50dp"
          />
```
Start does not override Left and you can see both sides have padding.

My implementation is consistent with this (although this may be a bug in AOSP: https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/view/View.java#17131 ?) and frankly the developer should not be mixing and matching, but if they do, then it's WYSIWYG. You can set a Start, and also set a Left and Left will always set on the left side regardless of RTL/LTR orientation, and Start will adjust accordingly. That seems more intuitive than Google's implementation on margin where they override it.


> In the case of RTL layout direction on Android 4.2+, the top left corner should be rounded if roundTopEnd==true && roundTopRight is either true or not set

I did the following:
```
      <com.facebook.drawee.view.SimpleDraweeView
          android:id="@+id/drawee_some_rtl"
          style="@style/DraweeRoundedCorners.Drawee"
          app:roundedCornerRadius="16dp"
          app:roundTopEnd="true"
          app:roundTopRight="false"
          />
```
and the top left corner was rounded in RTL mode for me. Did you see something different?

Reply:
Hi @adamc01, sorry for taking a while to get back to you.

OK, that sounds reasonable and after checking the source for View.java I agree that this is consistent with the platform behaviour (as confusing as that may be).

Great work! I'll get this merged now